Classes/common/CDTDatastore.m

```diff
@@ -150,6 +168,11 @@ - (NSUInteger)documentCount
 
 - (NSString *)name { return self.database.name; }
 
+- (id<CDTEncryptionKeyProvider>)copyEncryptionKeyProvider
+{
+    return [_encryptionKeyProvider copy];
+}
```

Comment:
Copying the key provider doesn't provide the guarantee you describe because the provider can get its key from elsewhere in a dynamic manner (e.g., network request). I'm therefore unsure that this precaution gains us something here, and ends up being a little confusing.